### DISCUSSION TITLE
Output change from objects in lists in custom scripts

### DISCUSSION THREAD

**Discussion by PaulR282** [netbox-community/netbox#17218] [General]
I noticed the output of `self.log_info(some_device)` differs from `self.log_info([some_device])`. The first outputs the object name while the second outputs `[]`. This is problematic when outputting an object within a list. I don't think it is intended and it worked some time ago.

**Comment by candlerb** [comment]
Netbox version?

You can't pass random types to python functions and expect them to work. In Netbox 4.0, log_info is [documented](https://netboxlabs.com/docs/netbox/en/stable/customization/custom-scripts/#logging) to take two arguments, the first being a message and the second being an object, both optional.

```
log_info(message=None, obj=None)
```

Therefore, these should work:

```
self.log_info(None, some_device)
self.log_info(obj=some_device)
```

Passing an object as the message *might* happen to work, because Netbox *might* implicitly call `str(...)` on that object. If you pass an array of objects, well it's down to the implementation what might happen.

You could try using a function which explicitly converts to string form, e.g.

```
log_info(repr([some_device]))
```

**Reply by PaulR282** [reply]
Yes, I could use a function but I have a lot of structures with nested lists in a lot of scripts where I would need to implement this. Some time ago, it worked normally (outputting the `object.name`). I don't think it came with the v4 update, I think the change was more recent.
